src/auth.ts

```diff
@@ -169,85 +182,183 @@ export function initBaseAuth(opts: BaseAuthOptions) {
         fetchOrgByQuery: fetchOrgsByQueryWrapper,
         fetchUsersByQuery: fetchUsersByQueryWrapper,
         fetchUsersInOrg: fetchUsersInOrgWrapper,
+        // user management functions
         createUser: createUserWrapper,
         updateUserMetadata: updateUserMetadataWrapper,
         updateUserEmail: updateUserEmailWrapper,
         createMagicLink: createMagicLinkWrapper,
         migrateUserFromExternalSource: migrateUserFromExternalSourceWrapper,
-        createOrg: createOrgWrapper,
-        addUserToOrg: addUserToOrgWrapper,
         deleteUser: deleteUserWrapper,
         disableUser: disableUserWrapper,
         enableUser: enableUserWrapper,
+        // org management functions
+        createOrg: createOrgWrapper,
+        addUserToOrg: addUserToOrgWrapper,
         allowOrgToSetupSamlConnection: allowOrgToSetupSamlConnectionWrapper,
         disallowOrgToSetupSamlConnection: disallowOrgToSetupSamlConnectionWrapper,
-        UserRole,
     }
 }
 
+// wrapper function with no validation
 function wrapValidateAccessTokenAndGetUser(tokenVerificationMetadataPromise: Promise<TokenVerificationMetadata | void>) {
     return async function validateAccessTokenAndGetUser(authorizationHeader?: string): Promise<User> {
         return extractAndVerifyBearerToken(tokenVerificationMetadataPromise, authorizationHeader)
     }
 }
 
-function wrapValidateAccessTokenAndGetUserWithOrg(tokenVerificationMetadataPromise: Promise<TokenVerificationMetadata | void>) {
-    const validateAccessTokenAndGetUserWithOrgInfo = wrapValidateAccessTokenAndGetUserWithOrgInfo(tokenVerificationMetadataPromise)
+// The following eight functions are wrappers around our four validations: isRole, atLeastRole, hasRequirement, hasAllRequirements
+// There are two wrappers for each validation, depending on if you want to validate the orgId or orgInfo
+
+function wrapValidateAccessTokenAndGetUserWithOrgWithMinimumRole(tokenVerificationMetadataPromise: Promise<TokenVerificationMetadata | void>) {
+    const validateAccessTokenAndGetUserWithOrgInfo = wrapValidateAccessTokenAndGetUserWithOrgInfoWithMinimumRole(tokenVerificationMetadataPromise)
+    return async function validateAccessTokenAndGetUserWithOrgId(authorizationHeader: string | undefined,
+                                                                 requiredOrgId: string,
+                                                                 requiredRole?: string): Promise<UserAndOrgMemberInfo> {
```

Comment:
The original intention behind making minimumRequiredRole an optional parameter was because there was only one validate function - so it was basically choosing between:
- Validate the user is in the organization, and that's it
- Validate the user is in the organization, and they have at least that role

Now that there are 4 functions here, all of them being optional arguments feels a little strange. I'm not sure what the best path forward is, what do you think about a 5th function for just verifying org membership with no role/permission checking?

Reply:
That sounds good.